src/python/pants/core/util_rules/external_tool.py

```diff
@@ -119,6 +120,34 @@ def register_options(cls, register):
             advanced=True,
             help=help_str,
         )
+        register(
+            "--url-template",
+            type=str,
+            default=cls.default_url_template,
+            advanced=True,
+            help=(
+                "URL to download the tool, either as a single binary file or a compressed file "
+                "(e.g. zip file). You can change this to point to your own hosted file, e.g. to "
+                "work with proxies. Use {version} to have the value from --version substituted, "
+                "and {platform} to have a value from --url-platform-mapping substituted in, "
+                "depending on the current platform. "
+                "For example, https://github.com/.../protoc-{version}-{platform}.zip."
+            ),
+        )
+        register(
+            "--url-platform-mapping",
+            type=dict,
+            default=cls.default_url_platform_mapping,
+            advanced=True,
+            help=(
+                "A dictionary mapping platforms to strings to be used when generating the URL "
```

Comment:
Other reviewers, this is the best that Thales and I could come up with, but please feel free to tweak. It's dense.